projects/grafana/setup-grafana.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/sh
```

Comment:
Suggest script like:

```sh
#!/bin/bash

datasource_name=$1
prometheus_namespace="$2"

oc new-project grafana
oc new-app -f grafana-ocp.yaml
oc rollout status deployment/grafana-ocp

payload="$( mktemp )"
cat <<EOF >"${payload}"
{
	"name": "${datasource_name}",
	"type": "prometheus",
	"typeLogoUrl": "",
	"access": "proxy",
	"url": "https://$( oc get route prometheus -n "${prometheus_namespace}" -o jsonpath='{.spec.host}' )",
	"basicAuth": false,
	"withCredentials": false,
	"jsonData": {
		"tlsSkipVerify":true,
		"token":"$( oc sa get-token grafana-ocp )"
	}
}
EOF

grafana_host="$( oc get route grafana-ocp -o jsonpath='{.spec.host}' )"
curl -H "Content-Type: application/json" -u admin:admin "${grafana_host}/api/datasources" -X POST -d "@${payload}"

dashboard_file="./openshift-cluster-monitoring.json"
sed -i.bak "s/\${DS_PR}/${datasource_name}/" "${dashboard_file}"
curl -H "Content-Type: application/json" -u admin:admin "${grafana_host}/api/dashboards/db" -X POST -d "@${dashboard_file}"
mv "${dashboard_file}.bak" "${dashboard_file}"
```

Reply:
good suggestion.
could you explain those items:
 - why do you want to pass the prom namesapace as arg, I think is better to grab it dynamically
- why do we need a rollout?
- we probably still need the loop that waits grafana pods to be running.

Reply:
> why do you want to pass the prom namesapace as arg, I think is better to grab it dynamically

Otherwise you need cluster-reader or larger scope to figure out where the thing is. Could even pass in the hostname

> why do we need a rollout?

You were already doing one -- just not correctly. Polling for pod status is error prone. The nice UX from `oc` for this is to use `oc rollout status`

> we probably still need the loop that waits grafana pods to be running.

see above